### DISCUSSION TITLE
no-cond-assign seems to have except-parens disabled and turning them back on throws linting errors.

### DISCUSSION THREAD

**Discussion by m-shum** [antfu/eslint-config#468] [Q&A]
### Describe the bug

no-cond-assign rule has an except-parens option. This appears to be disabled in this config, which means code like this: `while ((m = regex.exec(str)) !== null)` throws lint errors.

I have tried enabling it using`'no-cond-assign': ['error', 'except-parens'],` but the config auto-changes this to `'no-cond-assign': ['error', {'except-parens': true}]`, which also throws a lint error as it's incorrect syntax.

### Reproduction

https://stackblitz.com/edit/github-lzpjei-ccwgoe?file=.vscode%2Fsettings.json,app.vue,eslint.config.js

### System Info

```Shell
System:
    OS: macOS 14.4.1
    CPU: (8) arm64 Apple M2
    Memory: 101.30 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
  Browsers:
    Chrome: 124.0.6367.118
    Safari: 17.4.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by antfu** [comment]
Thank you for submitting the issue!

However, please also note that this is only a shared config where we can only make sure the default and recommended configs are working fine. We don't have the bandwidth to deal with custom configuration or rules conflicts. If you believe it's a bug in the rules, please create an issue to upstream plugins instead.

Thus, I am converting this to a discussion for now.

**Reply by m-shum** [reply]
Understandable, of course!
My main concern here is that I'm actually unable to turn on `except-parens`. As I mentioned, any time I add that rule, it becomes automatically transformed into` ['string', {object}] `syntax, which this rule does not accept.
